delivery_cttexpress/models/delivery_carrier.py

```diff
@@ -0,0 +1,310 @@
+        """
+        status = "{} - [{}] {}".format(
+            fields.Datetime.to_string(tracking["StatusDateTime"]),
+            tracking["StatusCode"],
+            tracking["StatusDescription"],
+        )
+        if tracking["IncidentCode"]:
+            status += " ({}) - {}".format(
+                tracking["IncidentCode"], tracking["IncidentDescription"]
+            )
+        return status
+
+    @api.onchange("cttexpress_shipping_type")
+    def _onchange_cttexpress_shipping_type(self):
+        """Control service validity according to credentials
+
+        :raises UserError: We list the available services for given credentials
+        """
+        if not self.cttexpress_shipping_type:
+            return
+        # Avoid checking if credentianls aren't setup or are invalid
+        try:
+            self.action_ctt_validate_user()
+        except UserError:
+            return
+        ctt_request = self._ctt_request()
+        error, service_types = ctt_request.get_service_types()
+        self._ctt_log_request(ctt_request)
+        self._ctt_check_error(error)
+        type_codes, type_descriptions = zip(*service_types)
+        if self.cttexpress_shipping_type not in type_codes:
+            service_name = dict(
+                self._fields["cttexpress_shipping_type"]._description_selection(
+                    self.env
+                )
+            )[self.cttexpress_shipping_type]
+            raise UserError(
+                _(
+                    "This CTT Express service (%s) isn't allowed for this account "
+                    "configuration. Please choose one of the followings\n%s"
+                    % (service_name, type_descriptions)
+                )
+            )
+
+    def action_ctt_validate_user(self):
+        """Maps to API's ValidateUser method
+
+        :raises UserError: If the user credentials aren't valid
+        """
+        self.ensure_one()
+        ctt_request = self._ctt_request()
+        error = ctt_request.validate_user()
+        self._ctt_log_request(ctt_request)
+        # For user validation success there's an error return as well.
+        # We better ignore it.
+        if error[0]:
+            self._ctt_check_error(error)
+
+    def _prepare_cttexpress_shipping(self, picking):
+        """Convert picking values for CTT Express API
+
+        :param record picking: `stock.picking` record
+        :return dict: Values prepared for the CTT connector
+        """
+        self.ensure_one()
+        # A picking can be delivered from any warehouse
+        sender_partner = (
+            picking.picking_type_id.warehouse_id.partner_id
+            or picking.company_id.partner_id
+        )
+        recipient = picking.partner_id
+        recipient_entity = picking.partner_id.commercial_partner_id
+        if picking.package_ids:
+            weight = sum([p.shipping_weight or p.weight for p in picking.package_ids])
+        else:
+            weight = picking.shipping_weight
```

Comment:
```suggestion
        weight = picking.shipping_weight
```

Reply:
Since v14 shipping_weight field is correctly (https://github.com/odoo/odoo/blob/14.0/addons/delivery/models/stock_picking.py#L70)

Reply:
Fixed. Thanks for the heads on!